File: contest/ICPC/2018/欧洲/NWERC西北欧/data/kleptography/submissions/accepted/AR.java
// Author Alexander Rass
import java.util.*;

public class AR{
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        int N = scan.nextInt(),M = scan.nextInt();
        String last_part = scan.next(), encrypted = scan.next();
        char[] plain = new char[M];
        for(int i = 0; i < N; i++)
            plain[M-N+i] = last_part.charAt(i);
        for(int i = M-N-1; i >= 0; i--)plain[i] = (char)((((encrypted.charAt(i+N) - 'a') - (plain[i+N] - 'a') + 26) % 26) + 'a');
        for(int i = 0; i < M; i++)System.out.print(plain[i]);
        System.out.println();
    }
}

File: contest/ICPC/2018/欧洲/NWERC西北欧/data/harddrive/submissions/accepted/ar.cpp
// Author Alexander Rass
#include <bits/stdc++.h>
using namespace std;

int main(){
    int N,AIM,BROKEN;
    cin >> N >> AIM >> BROKEN;
    int DONE = 1;
    if(AIM&1){
        printf("10");
        DONE = 2;
    } else {
        printf("0");
    }
    AIM /= 2;
    for(;BROKEN;--BROKEN){
        int NEXT;
        cin >> NEXT;
        while(AIM > 0 && NEXT - DONE >= 2){
            printf("10");
            DONE+=2;
            --AIM;
        }
        while(NEXT != DONE){
            printf("0");
            ++DONE;
        }
    }
    printf("\n");
	return 0;
}
